test/e2e/service.go

```diff
@@ -740,6 +740,7 @@ var _ = ginkgo.Describe("Services", func() {
 					}
 				}
 			}
+			nodeIPs = make(map[string]map[int]string)
```

Comment:
Shouldn't we do this also just before the `framework.Failf("failed to remove ip address %s from node %s, err: %q", ip, nodeName, err)` call above?

Reply:
I meant when we fail.  `framework.Failf(..)` will break out of the loop, right?  But if we don't reset the map there we might cause other tests in the suite (that are executed after the currently failing one) to fail too.

Reply:
came here to post the same fix :)
I think the most robust way would be to move 
```
nodeIPs = make(map[string]map[int]string)
// and maybe this one for company
targetSecondaryNode = node{
			name: "egressSecondaryTargetNode-allowed",
		}
```
to BeforeEach?

Reply:
Good point, I will wait for the control-plane tests to pass and will change this to:
```
ginkgo.BeforeEach(func() {
	nodeIPs = make(map[string]map[int]string)
})
```
I think it makes more sense logically anyway.

Reply:
Actually, I will reset all shared vars to avoid any future issues.

Reply:
All control-plane lanes are green, re-pushing.